basyx.submodelrepository/basyx.submodelrepository-backend-inmemory/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/InMemorySubmodelRepository.java

```diff
@@ -220,8 +224,18 @@ public SubmodelValueOnly getSubmodelByIdValueOnly(String submodelId) {
 	@Override
 	public Submodel getSubmodelByIdMetadata(String submodelId) {
 		Submodel submodel = getSubmodel(submodelId);
-		submodel.setSubmodelElements(null);
-		return submodel;
+		try {
+			String submodelAsJSON = new JsonSerializer().write(submodel);
+			Submodel submodelDeepCopy = new JsonDeserializer().readReferable(submodelAsJSON, Submodel.class);
+			submodelDeepCopy.setSubmodelElements(null);
+			return submodelDeepCopy;
+		} catch (DeserializationException e) {
+			e.printStackTrace();
```

Comment:
Please pass this exception as additional argument to the RuntimeException instead of printing the stacktrace here.